pkg/rhcos/ami.go

```diff
@@ -10,14 +16,72 @@ const (
 )
 
 // AMI calculates a Red Hat CoreOS AMI.
-func AMI(channel, region string) (ami string, err error) {
+func AMI(ctx context.Context, channel, region string) (ami string, err error) {
 	if channel != DefaultChannel {
 		return "", fmt.Errorf("channel %q is not yet supported", channel)
 	}
 
-	if region != "us-east-1" {
-		return "", fmt.Errorf("region %q is not yet supported", region)
+	ssn := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+		Config: aws.Config{
+			Region: aws.String(region),
+		},
+	}))
+
+	svc := ec2.New(ssn)
+
+	result, err := svc.DescribeImagesWithContext(ctx, &ec2.DescribeImagesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("name"),
+				Values: aws.StringSlice([]string{"rhcos*"}),
+			},
+			{
+				Name:   aws.String("architecture"),
+				Values: aws.StringSlice([]string{"x86_64"}),
+			},
+			{
+				Name:   aws.String("virtualization-type"),
+				Values: aws.StringSlice([]string{"hvm"}),
+			},
+			{
+				Name:   aws.String("image-type"),
+				Values: aws.StringSlice([]string{"machine"}),
+			},
+			{
+				Name:   aws.String("owner-id"),
+				Values: aws.StringSlice([]string{"531415883065"}),
+			},
+			{
+				Name:   aws.String("state"),
+				Values: aws.StringSlice([]string{"available"}),
+			},
+		},
+	})
+	if err != nil {
+		return "", err
+	}
+
+	var image *ec2.Image
+	var created time.Time
+	for _, nextImage := range result.Images {
+		if nextImage.ImageId == nil || nextImage.CreationDate == nil {
+			continue
+		}
+		nextCreated, err := time.Parse(time.RFC3339, *nextImage.CreationDate)
+		if err != nil {
+			return "", err
+		}
+
+		if image == nil || nextCreated.Before(created) {
```

Comment:
I think I have the sign wrong here; we want the most recent AMI, not the oldest.

/hold

Reply:
> I think I have the sign wrong here...

Fixed with f561c2e -> 801c21f.

/hold cancel